src/http_protocol.cpp

```diff
@@ -731,4 +754,42 @@ std::string http_response_t::generic(unsigned code,
   return response;
 }
 
-} // namespace prime_server
+// namespace prime_server
+}
+
+// make a short circuiter that will respond to health checks and OPTIONS request
+// Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
+prime_server::shortcircuiter_t<prime_server::http_request_t> make_shortcircuiter(const uint8_t& verb_mask = std::numeric_limits<uint8_t>::max(),
+                                                                                      const std::string& health_check_path = "/health_check") {
+
+  std::string allowed_verbs = get_allowed_methods_string(verb_mask);
+
+// TODO: Answer to preflight requests in a more robust way 
+  return [verb_mask, allowed_verbs, health_check_path](const prime_server::http_request_t& request) {
```

Comment:
you can just capture by value all the stuff rather than listing it out

```suggestion
  return [=](const prime_server::http_request_t& request) {
```